Add Xamarin troubleshooting (#3878)
getsentry/sentry-docs: src/platforms/dotnet/guides/xamarin/common/troubleshooting.mdx

> ---

Reviewer: This file isn't displaying in preview. Revise it to `common/troubleshooting/index.mdx`.

Author: Moving to `common/troubleshooting/index.mdx` didn't help.
But leaving the doc at the original place and adding troubleshooting file to .NET fixed it, but then, the .NET doc ended up with a blank troubleshooting page while the Xamarin one got proper troubleshooting 😅.
So I believe it'll be fine to leave all the Xamarin troubleshooting info into its own index file.

Reviewer: Something is unfortunately awry here. The troubleshooting information is now displaying on the Getting Started page, which is not the way our content works. The information architecture is designed to get users up and running (the primary index page for the platform), with sections then available for configuration, usage, and other key elements (such as source maps for JS). 

We can't put the troubleshooting information on the primary index page of the platform; it should be its own page. I'm happy to help you if you get stuck enabling this.

Author: I accept some help @PeloWriter 😄
Long short story, I was trying to follow the layout chosen for Unity, https://docs.sentry.io/platforms/unity/troubleshooting/,
 but Unity is a `platform` and not a `guide` so things got tricky for me.

title:  Troubleshooting
sidebar_order: 100
description: "Solutions to common issues"
---

> 

Reviewer: suggest adding an intro here:
"If you need help solving issues with your Sentry's Xamarin SDK, you can read the edge cases documented here. If you need additional help, you can view our forums, and customers on a paid plan may also contact support."

Author: sounds great :D

## Offline Caching Information 

> 

Reviewer: Going directly from one heading level to the next (without text between) interferes with the RH sidebar. Perhaps add something similar to:
"For users on Xamarin iOS, offline caching may cause this tricky issue."

### Event/Transaction Capture - `Failed to send cached envelope: System.MissingMethodException: Method not found: System.Text.Json.JsonDocument`

> If the above message appears for you when capturing an event or transaction on your debug window, you'll have to edit your have to manually edit your platform csproj 

Reviewer: ```suggestion
If this message displays when capturing an event or transaction on your debug window, manually edit your platform `csproj`:
```

> 1. close your IDE
> 2. open your iOS .csproj with an text/xml editor
> 3. add the following nuget reference:

Reviewer: ```suggestion
1. Close your IDE
2. Ppen your iOS `.csproj` with a text or xml editor
3. Add the following NuGet reference:
```

```xml
    <PackageReference Include="System.Memory" IncludeAssets="None">
      <Version>4.5.4</Version>
    </PackageReference>
```  

> NOTE: it's important to add the IncludeAssets as None, otherwise this fix will not work.

Reviewer: ```suggestion
<Note>

It's important to add the `IncludeAssets` as `None`, otherwise this fix will not work.

</Note>
```

> 4. save and open your IDE
> 5. wait for it to restore the packages and once finished try to capture an exception or transaction.

Reviewer: ```suggestion
4. Save and open your IDE
5. Wait for the IDE to restore the packages. Once finished, try to capture an exception or transaction.
```

> Another way to validate if the problem is happening on an iOS project, is by adding the following code to your iOS project and check if there's any conflict:

Reviewer: ```suggestion
Another way to validate if the problem is happening on an iOS project is by adding the following code to your iOS project to check if there's a conflict:
```
